packages/yourloops/test/integration/assert/caregivers-list.assert.ts

```diff
@@ -0,0 +1,184 @@
+
+  const inviteButton = within(addCaregiverDialog).getByRole('button', { name: 'Invite' })
+  expect(inviteButton).toBeVisible()
+  expect(inviteButton).toBeDisabled()
+
+  const emailInput = within(addCaregiverDialog).getByRole('textbox', { name: 'Email' })
+  const validEmail = 'valid-email@test.com'
+  await userEvent.type(emailInput, validEmail)
+  expect(inviteButton).toBeEnabled()
+
+  await userEvent.clear(emailInput)
+  expect(inviteButton).toBeDisabled()
+
+  await userEvent.type(emailInput, 'invalid-email')
+  expect(inviteButton).toBeDisabled()
+  expect(within(addCaregiverDialog).getByText('Invalid email address (special characters are not allowed).')).toBeVisible()
+
+  await userEvent.clear(emailInput)
+  await userEvent.type(emailInput, validEmail)
+  expect(inviteButton).toBeEnabled()
+  expect(within(addCaregiverDialog).queryByText('Invalid email address (special characters are not allowed).')).not.toBeInTheDocument()
+
+  await userEvent.clear(emailInput)
+  await userEvent.type(emailInput, mockCaregiverUser.username)
+  expect(inviteButton).toBeDisabled()
+  expect(within(addCaregiverDialog).getByText('You are already sharing your data with this caregiver.')).toBeVisible()
+
+  await userEvent.clear(emailInput)
+  await userEvent.type(emailInput, 'other-patient@mail.com')
+
+  jest.spyOn(DirectShareApi, 'addDirectShare').mockRejectedValueOnce(new Error(PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_MESSAGE))
+  await userEvent.click(inviteButton)
+
+  expect(DirectShareApi.addDirectShare).toHaveBeenCalled()
+  expect(screen.getByTestId('alert-snackbar')).toHaveTextContent('You cannot share your data with this user as they are not a caregiver.')
+  await userEvent.click(within(screen.getByTestId('alert-snackbar')).getByTitle('Close'))
+  expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
+}
+
+export const checkRemoveCaregiverCancel = async (caregiverEmail: string) => {
+  const caregiversTable = screen.getByTestId('patient-caregivers-list')
+  const caregiverRow = within(caregiversTable).getByTestId(`patient-caregivers-table-row-${caregiverEmail}`)
+  const removeCaregiverButton = within(caregiverRow).getByTestId(`patient-caregivers-table-row-${caregiverEmail}-button-remove`)
+
+  await userEvent.click(removeCaregiverButton)
+
+  const removeCaregiverDialog = screen.getByRole('dialog')
+  const cancelButton = within(removeCaregiverDialog).getByRole('button', { name: 'Cancel' })
+  expect(cancelButton).toBeEnabled()
+
+  await userEvent.click(cancelButton)
+
+  expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
+}
+
+export const checkRemoveCaregiverSuccess = async (caregiverEmail: string) => {
+  const caregiversTable = screen.getByTestId('patient-caregivers-list')
+  const caregiverRow = within(caregiversTable).getByTestId(`patient-caregivers-table-row-${caregiverEmail}`)
+  const removeCaregiverButton = within(caregiverRow).getByTestId(`patient-caregivers-table-row-${caregiverEmail}-button-remove`)
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏Same